.github/workflows/test.main.kts

```diff
@@ -117,17 +131,62 @@ private fun validateTypings() {
         { it.owner == "DamianReeves" && it.name == "write-file-action" },
     )
 
+    println()
+    val actions = baseRef.let { baseRef ->
+        if (baseRef == null) {
+            println("Validating all typings")
 
-    val actionsWithYamlExtension = Files.walk(Path("typings"))
-        .filter { it.name == "action-types.yaml" }
-        .toList()
-    check(actionsWithYamlExtension.isEmpty()) {
-        "Some files have .yaml extension, and we'd like to use only .yml here: $actionsWithYamlExtension"
-    }
+            val actionsWithYamlExtension = Files.walk(Path("typings"))
+                .filter { it.name == "action-types.yaml" }
+                .toList()
+            check(actionsWithYamlExtension.isEmpty()) {
+                "Some files have .yaml extension, and we'd like to use only .yml here: $actionsWithYamlExtension"
+            }
 
-    val actions = Files.walk(Path("typings"))
-        .filter { it.name == "action-types.yml" }
-        .map {
+            Files.walk(Path("typings")).filter { it.name == "action-types.yml" }
+        } else {
+            println("Only validating changed typings")
+
+            val typings = try {
+                KGit.open(File(".")).use { git ->
+                    git.fetch {
+                        setRefSpecs("refs/heads/$baseRef:refs/heads/$baseRef")
+                        setDepth(1)
+                    }
+                    git.diff {
+                        setShowNameAndStatusOnly(true)
+                        git.repository.newObjectReader().use { objectReader ->
+                            setOldTree(CanonicalTreeParser().apply {
+                                reset(objectReader, git.repository.resolve("refs/heads/$baseRef^{tree}"))
+                            })
+                            setNewTree(CanonicalTreeParser().apply {
+                                reset(objectReader, git.repository.resolve("$sha^{tree}"))
+                            })
+                        }
+                        setPathFilter(
+                            AndTreeFilter.create(
+                                PathFilter.create("typings/"),
+                                OrTreeFilter.create(
+                                    PathSuffixFilter.create("/action-types.yml"),
+                                    PathSuffixFilter.create("/action-types.yaml"),
+                                ),
+                            )
+                        )
+                    }
+                        .filter { it.changeType != DELETE }
+                        .map { Path(it.newPath) }
+                        .groupBy { it.extension == "yml" }
+                }
+            } finally {
+                KGit.shutdown()
```

Comment:
nit: too bad we need it, despite using `.use { ... }` 🤔

Reply:
Yeah, well, `use` just calls `close` on the `KGit` instance and that `shutdown` does clean up other stuff, so I just assumed this should be done for an orderly shutdown. 🤷‍♂️